File: golden/javascript/longest_common_prefix.js
function solution(words) {
    if (!words.length) {
        return '';
    }
    if (words.length == 1) {
        return words[0];
    }

    var prefix = words[0];
    words.forEach((word) => {
        newPrefix = '';
        for (var i = 0; i < Math.min(prefix.length, word.length); i++) {
            if (prefix[i] != word[i]) {
                break;
            }
            newPrefix += word[i];
        }
        prefix = newPrefix;
    });
    return prefix;
}

File: responses/TK75Q6-7YJ/longest_common_prefix.py
def solution(A):
    index = 0
    result = ""
    while True:
        currentChar = "!"
        for word in A:
            if len(word) <= index:
                return result
            if currentChar == "!":
                currentChar = word[index]
                continue
            if currentChar != word[index]:
                return result
        result += currentChar
        index += 1
    return result
